src/App.tsx

```diff
@@ -1,7 +1,46 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressKey: string;
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    pressKey: '',
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', this.handleKeyPress);
+  }
+
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', this.handleKeyPress);
+  }
+
+  private handleKeyPress = (event: KeyboardEvent) => {
+    this.setState({
+      pressKey: event.key,
+    });
+  };
+
+  render() {
+    const { pressKey } = this.state;
+
+    return (
+      <div className="App">
+        {pressKey ? (
+          <p className="App__message">
+            The last pressed key is
+            [
+            {pressKey}
+            ]
+          </p>
+        ) : (
+          <p className="App__message">
+            Nothing was pressed yet
+          </p>
+        )}
```

Comment:
You do not need to write two identical <p> tags - it is enough to write this tag once and move the condition for the message inside it.

Reply:
I agree!